### ISSUE TITLE
php bakery build-assets --compile Failing v 4.4.3

### ISSUE THREAD

**Issue by pauljcurtis** [userfrosting/UserFrosting#1101]
I've reported this issue on the UF chat. This is failing on using puTTY on a Windows 10 PC and directly on cPanel. Apache setup.

Running `php bakery build-assets --compile` fails with the following error.

```(node:56307) UnhandledPromiseRejectionWarning: TypeError: Cannot set property 'path' of undefined
    at VinylFsVPathSrc._read (/UserFrosting/build/node_modules/@userfrosting/vinyl-fs-vpath/dist/src.js:46:27)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
(node:56307) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:56307) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js processwith a non-zero exit code.
[15:35:28] The following tasks did not complete: build
[15:35:28] Did you forget to signal async completion?
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ uf-bundle: `gulp build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ uf-bundle script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mysite/.npm/_logs/2020-06-16T14_35_28_251Z-debug.log```

[2020-06-16T14_35_28_251Z-debug.log](https://github.com/userfrosting/UserFrosting/files/4787815/2020-06-16T14_35_28_251Z-debug.log)

**Comment by pauljcurtis** [comment]
I've just ran `php bakery build-assets --compile` again to triple check. It seems to fail after building different assets. There's no consistency.

```(node:105590) UnhandledPromiseRejectionWarning: TypeError: Cannot set property 'path' of undefined
    at VinylFsVPathSrc._read (/UserFrosting/build/node_modules/@userfrosting/vinyl-fs-vpath/dist/src.js:46:27)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
(node:105590) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:105590) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[18:11:23] The following tasks did not complete: build
[18:11:23] Did you forget to signal async completion?
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ uf-bundle: `gulp build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ uf-bundle script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.```

**Comment by lcharette** [comment]
Can't replicate on theses env : 

- MacOS, UF 4.3.3, Node v10.16.3, NPM 6.13.7
- MacOS, UF 4.4.3, Node v10.16.3, NPM 6.13.7
- Ubuntu (Homestead), UF 4.4.3, Node v12.17.0, NPM 6.14.4

**Comment by lcharette** [comment]
Can you run `php bakery debug` on all your env and share the result? We will be able to compare with the Node/NPM version too.

**Comment by Silic0nS0ldier** [comment]
Looking at the stacktrace, this issue most likely was caused by a permission issue. While there should have been a more helpful error message, having taken a look at the `@userfrosting/vinyl-fs-vpath` code, I've identified an implementation flaw.

https://github.com/userfrosting/vinyl-fs-vpath/issues/284

The fix here won't resolve your issue, but should point the finger in the right place at least.
